js/style/style_declaration.js

```diff
@@ -15,18 +15,21 @@ function StyleDeclaration(reference, value) {
     this.json = JSON.stringify(this.value);
 
     var parsedValue = this.type === 'color' ? parseColor(this.value) : value;
-    if (reference.function === 'interpolated') {
-        this.calculate = MapboxGLFunction.interpolated(parsedValue);
-    } else {
-        this.calculate = MapboxGLFunction['piecewise-constant'](parsedValue);
-        if (reference.transition) {
-            this.calculate = transitioned(this.calculate);
-        }
+    this.calculate = MapboxGLFunction[reference.function || 'piecewise-constant'](parsedValue);
+    this.isFeatureConstant = this.calculate.isFeatureConstant;
+    this.isGlobalConstant = this.calculate.isGlobalConstant;
+
+    if (reference.function === 'piecewise-constant' && reference.transition) {
+        this.calculate = transitioned(this.calculate);
     }
 }
 
 function transitioned(calculate) {
-    return function(z, zh, duration) {
+    return function(globalProperties, featureProperties) {
+        var z = globalProperties.$zoom;
+        var zh = globalProperties.$zoomHistory;
+        var duration = globalProperties.$duration;
```

Comment:
At one point, yes, but not since https://github.com/mapbox/mapbox-gl-function/pull/15. I will remove all `$` prefixed global properties.